src/pages/home/report/ReportActionCompose.js

```diff
@@ -379,7 +377,10 @@ class ReportActionCompose extends React.Component {
     insertSelectedEmoji(highlightedEmojiIndex) {
         const commentBeforeColon = this.state.value.slice(0, this.state.colonIndex);
         const emojiCode = this.state.suggestedEmojis[highlightedEmojiIndex].code;
-        const commentAfterColonWithEmojiNameRemoved = this.state.value.slice(this.state.colonIndex).replace(/^:\w+($|\s)/, ' ');
+
+        // Define the regular expression pattern to match a string starting with a colon and ending with a space or newline character
+        const pattern = /^:[^\n\r]+?(?=$|\s)/;
```

Comment:
NAB but there's a small amount of overhead with creating this regex anew every time this function is called. Maybe we should define it as a CONST somewhere else so that it's only created once?

Reply:
Yeah we should definitely put it in `CONST.js` https://github.com/Expensify/App/blob/a9d90e5a75263112faf7ef3db46f828362762088/src/CONST.js#L767-L802

Reply:
ok, this makes sense, I'll transfer them to const